src/Evolution/Systems/Cce/GaugeTransformBoundaryData.cpp

```diff
@@ -0,0 +1,685 @@
+    const tnsr::i<DataVector, 3>& cartesian_cauchy_coordinates,
+    const Scalar<SpinWeighted<ComplexDataVector, 0>>& omega,
+    const Scalar<SpinWeighted<ComplexDataVector, 1>>& eth_omega,
+    const size_t l_max) noexcept {
+  const size_t number_of_angular_points =
+      Spectral::Swsh::number_of_swsh_collocation_points(l_max);
+  const size_t number_of_radial_points =
+      get(*volume_u).size() / number_of_angular_points;
+
+  const ComplexDataVector u_scri_view;
+  make_const_view(make_not_null(&u_scri_view), get(*volume_u).data(),
+                  (number_of_radial_points - 1) * number_of_angular_points,
+                  number_of_angular_points);
+
+  get(*evolution_gauge_u_at_scri).data() = u_scri_view;
+
+  for (size_t i = 0; i < number_of_radial_points; ++i) {
+    ComplexDataVector angular_view_u{
+        get(*volume_u).data().data() +
+            i * Spectral::Swsh::number_of_swsh_collocation_points(l_max),
+        Spectral::Swsh::number_of_swsh_collocation_points(l_max)};
+    angular_view_u -= get(*evolution_gauge_u_at_scri).data();
+  }
+
+  Variables<
+      tmpl::list<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<1, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<2, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<3, ComplexDataVector>,
+                                      std::integral_constant<int, 1>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<4, ComplexDataVector>,
+                                      std::integral_constant<int, 1>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<5, ComplexDataVector>,
+                                      std::integral_constant<int, 1>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<6, ComplexDataVector>,
+                                      std::integral_constant<int, 2>>,
+                 ::Tags::SpinWeighted<::Tags::TempScalar<7, ComplexDataVector>,
+                                      std::integral_constant<int, 0>>>>
+      computation_buffers{number_of_angular_points};
+
+  auto& x =
+      get(get<::Tags::SpinWeighted<::Tags::TempScalar<0, ComplexDataVector>,
+                                   std::integral_constant<int, 0>>>(
+          computation_buffers));
+  auto& y =
+      get(get<::Tags::SpinWeighted<::Tags::TempScalar<1, ComplexDataVector>,
+                                   std::integral_constant<int, 0>>>(
+          computation_buffers));
+  auto& z =
+      get(get<::Tags::SpinWeighted<::Tags::TempScalar<2, ComplexDataVector>,
+                                   std::integral_constant<int, 0>>>(
+          computation_buffers));
+  x.data() =
```

Comment:
Add comment like `// Switch to complex so we can take spin-weighted derivatives`